src/components/ModalWizard/ModalWizardBody.js

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { bindMethods, noop, EmptyState, Spinner, Wizard } from '../../index';
+
+class ModalWizardBody extends React.Component {
+  constructor() {
+    super();
+    bindMethods(this, ['onStepClick', 'stepProps', 'renderLoading']);
+  }
+
+  onStepClick(stepIndex) {
+    const {
+      goToStep,
+      nextStepDisabled,
+      stepButtonsDisabled,
+      activeStepIndex
+    } = this.props;
+    // Don't allow step clicks to skip into the future, but skipping into the past is ok.
+    const nextStepClicked = stepIndex === activeStepIndex + 1;
+    const futureStepClicked = stepIndex > activeStepIndex + 1;
```

Comment:
is this piece too opinionated? I think we have the case we want to prevent the user from going back on Step 5 in v2v...

Reply:
There may be steps that are defaulted and can be skipped. I believe the application should decide if it is OK to navigate to a step.

Reply:
Yeah. I didn't really finish figuring this part out because this behavior is totally turned off in v2v. I think I am going to add back in the `disableGoto` piece that we had added and then abandoned in v2v, so that you can turn these buttons off step-by-step, and like you say, the application can decide.

Reply:
From Slack, my plan to address this flexibly is:

> i'll have props for `isValid`, `allowEnter` and `allowExit`. `allowEnter` will default to the `isValid` of the previous step, but you can override it if you need to. we can use `allowExit: false` on our final step.

Reply:
> is this piece too opinionated? I think we have the case we want to prevent the user from going back on Step 5 in v2v...

In VM provision wizard, it should be possible to skip steps (jump to a later step).

> There may be steps that are defaulted and can be skipped. I believe the application should decide if it is OK to navigate to a step.

Exactly my thoughts.

@mturley +1 on the proposed props for controlling the step navigation.

Reply:
Do you guys like these properties (`isValid`, `allowEnter`, `allowExit`) in the `steps` array? I think in the stateful wrapper version I might also offer `shouldBeValid(stepIndex)`, `shouldAllowEnter(stepIndex)` and `shouldAllowExit(stepIndex)` callbacks that you can pass in (without having to track the state and pass that in). But I can't decide if that is overkill.

Reply:
@vojtechszocs @jeff-phillips-18 @priley86 @AllenBW curious about your thoughts on the boolean props being available on the inner, stateless `WizardPattern` i.e:
```
<WizardPattern steps={[{ allowEnter: true, ... }, ...]} />
```
while on the stateful wrapper, we can pass something that will take the current step index as an argument (since when you're using the stateful wrapper, you probably are not tracking the current step) and still allow you to have logic based on the step index:
```
<StatefulWizardPattern steps={[{ shouldAllowEnter: (activeStepIndex) => { ... }, ... }, ...]} />
```

Of course, all the props passed to `StatefulWizardPattern` are passed down into the stateless `WizardPattern`, so if you want to use `allowEnter` explicitly from there, you can do that. So even if we did a stateful wizard for v2v, we could still just pass `allowExit: false` instead of having to write `shouldAllowExit: () => false`. Should make it easier for people to switch/upgrade/downgrade.

Reply:
I guess there is no reason for `shouldAllowExit` to take the current step index. and `shouldBeValid` doesn't make sense that it could change for step A when you move from step B to C. so really i guess it's just `shouldAllowEnter` that needs the `activeStepIndex` information, and the others shouldn't be callbacks.

Reply:
In either case, if you pass the simple booleans, that trumps any callback values. right?

Reply:
Maybe the entire thing should instead be one large `shouldStepChange(activeStepIndex, targetStepIndex)` callback at the top? that'll distinguish it from `onStepChanged` better too.

Reply:
Note: please don't reply here, I moved all of the above down to the main discussion thread for easy access.